File: db4oj.tests/src/com/db4o/db4ounit/common/persistent/Atom.java
/* Copyright (C) 2004   Versant Inc.   http://www.db4o.com */

package com.db4o.db4ounit.common.persistent;

public class Atom {
	
	public Atom child;
	public String name;
	
	public Atom(){
	}
	
	public Atom(Atom child){
		this.child = child;
	}
	
	public Atom(String name){
		this.name = name;
	}
	
	public Atom(Atom child, String name){
		this(child);
		this.name = name;
	}
	
	public int compareTo(Object obj){
		return 0;
	}
	
	public boolean equals(Object obj){
		if(obj instanceof Atom){
			Atom other = (Atom)obj;
			if(name == null){
				if(other.name != null){
					return false;
				}
			}else{
				if(! name.equals(other.name)){
					return false;
				}
			}
			if(child != null){
				return child.equals(other.child);
			}
			return other.child == null;
		}
		return false;
	}
	
	public String toString(){
		String str = "Atom(" + name + ")";
		if(child != null){
			return str + "." + child.toString();
		}
		return str;
	}
	
}

File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Persistent/Atom.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

namespace Db4objects.Db4o.Tests.Common.Persistent
{
	public class Atom
	{
		public Db4objects.Db4o.Tests.Common.Persistent.Atom child;

		public string name;

		public Atom()
		{
		}

		public Atom(Db4objects.Db4o.Tests.Common.Persistent.Atom child)
		{
			this.child = child;
		}

		public Atom(string name)
		{
			this.name = name;
		}

		public Atom(Db4objects.Db4o.Tests.Common.Persistent.Atom child, string name) : this
			(child)
		{
			this.name = name;
		}

		public virtual int CompareTo(object obj)
		{
			return 0;
		}

		public override bool Equals(object obj)
		{
			if (obj is Db4objects.Db4o.Tests.Common.Persistent.Atom)
			{
				Db4objects.Db4o.Tests.Common.Persistent.Atom other = (Db4objects.Db4o.Tests.Common.Persistent.Atom
					)obj;
				if (name == null)
				{
					if (other.name != null)
					{
						return false;
					}
				}
				else
				{
					if (!name.Equals(other.name))
					{
						return false;
					}
				}
				if (child != null)
				{
					return child.Equals(other.child);
				}
				return other.child == null;
			}
			return false;
		}

		public override string ToString()
		{
			string str = "Atom(" + name + ")";
			if (child != null)
			{
				return str + "." + child.ToString();
			}
			return str;
		}
	}
}
